creating-pull-requests.md

```diff
@@ -0,0 +1,131 @@
+# Creating a Pull Request
+
+This page goes over the steps for creating a Pull Request (PR), including some dos and don’ts.
+
+Note: the terms _Pull Request_ and _Merge Request_ can be used interchangeably. The same is true for the terms _issue_, _ticket_, and _stories_.
+
+## Before you start working
+
+Make sure that there is an issue created and assigned to you. Work on a feature branch that is named accordingly. It might also be a good idea to validate the approach you intend to take in case it is a big or complex issue.
+
+## When to create a Pull Request?
+
+The two main reasons to create a pull request are when your work is complete, or when you want early feedback on your work.
```

Comment:
I think a PR needs to get a code review before being approved. I am not sure if a PR is meant for code review/comments as form of getting feedback/input.

Reply:
> when you want early feedback on your work.

This is what Github calls **Draft** PRs, neither Assembla or bitbucket have this feature. This explicitly tells reviewers that _"hey, this PR is still WIP, but you can already start checking it out"_. Quite useful for large features or even experiments.

For a bitbucket project, I've prefixed some of my experiment PRs with **DRAFT*** to sort of mimic this. Just helps get the WIP intention across and prevent _"hey, you missed this obvious thing"_ in PR comments.